.github/workflows/run_tests.yaml

```diff
@@ -0,0 +1,130 @@
+name: Run tests
+
+on:
+
+  # Trigger the workflow on push but only for the master branch
+  push:
+    branches:
+      - master
+
+  # Trigger current workflow on PR from ANY branch
+  pull_request:
+    branches:
+      - '*'
+
+  # Cron Trigger
+  schedule:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule
+    # i.e. this runs every day at 4:33AM for the master branch.
+    - cron: '33 4 * * *'
+
+env:
+  # Even when given -y, apt will still sometimes hang at a prompt if a package
+  # has clarifications to ask; DEBIAN_FRONTEND=noninteractive prevents that,
+  # This will be defined for non-debian platforms below too, but there's no harm in that.
+  # (TravisCI quietly defined this on all their platforms, but we have to give it manually on GithubCI.)
+  DEBIAN_FRONTEND: 'noninteractive'
+
+jobs:
+  ultra_matrix_test:
+    name: Matrix Test of Python ${{ matrix.python-version }} on ${{ matrix.os }}
+
+    # Matrix driven OS
+    runs-on: ${{ matrix.os }}
+
+    # Default shell for ALL subsequent steps.
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    # Defining matrix for OS and Python
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ "ubuntu-latest", "macos-latest", "windows-latest" ]
+        python-version: [ "3.7", "3.8", "3.9" ]
+
+    # Main steps for the test to be reproduced across OS x Python
+    steps:
+      # Step 0: Checkout code.
+      - uses: actions/checkout@v2
+
+      # Step 1: List main repo by time modified, and print path
+      - name: List all current repository files and modification date
+        # Parameters.
+        run: |
+          ls -lasth
+          pwd
+
+      # Step 2: Install environment.
+      - name: Install Conda Env All OS
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{matrix.python-version}}
+          activate-environment: ads_venv
+          auto-update-conda: true
+          environment-file: environment.yml
+
+      # Step 3: List conda info, conda packages
+      - name: Conda info and list
+        # List
+        run: |
+          conda info
+          conda list
+
+      # Step 4: Activate and Install main AxonDeepSeg repository
+      - name: Installation of the main ADS repository
+        run: |
+          conda activate ads_venv
+          pip install --verbose -e .
+
+      # Step 5: Conda list packages POST ADS.
+      - name: Conda list post installation
+        # List
+        run: |
+          conda info
+          conda list
+
+      # Step 6: Flake 8 Linting
+      - name: Linting with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          pip install flake8
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Comment:
Small suggestion: In SCT's CI setup, we have linting as a separate job. Some motivations for this:

* Linting doesn't change from platform to platform, so including it in the build matrix might be unnecessary.
* If linting interrupts the test suite, you won't get any pytest feedback until after the linting issues are fixed, which can be a bit tedious. (Currently, we would have to commit -> check CI -> commit -> check CI more often due to linting issues.)

Reply:
@dyt811 I notice you removed this linting step. For reference, here's SCT's [workflow file for linting](https://github.com/neuropoly/spinalcordtoolbox/blob/master/.github/workflows/lint_code.yml) if you'd like to experiment.

We're currently using [Lintly](https://github.com/grantmcconnaughey/Lintly), which takes flake8 output and turns it into annotations on PRs.